### PR TITLE
Add missing Intel hardware intrinsics and fix names

### PR DESCRIPTION
**Pull request by fiigii**
Match the CoreCLR change https://github.com/dotnet/coreclr/pull/15601 
Have tested with the CoreCLR build.

cc @eerhardt @jkotas @tannergooding

### CODE REVIEW

File: src/System.Runtime.Intrinsics.X86/ref/System.Runtime.Intrinsics.X86.cs [hunk 1]
```diff
@@ -780,24 +813,27 @@ public static class Sse
...
+        public static Vector128<float> SetZeroVector128() { throw null; }
         public static Vector128<U> StaticCast<T, U>(Vector128<T> value) where T : struct where U : struct { throw null; }
         public static Vector128<float> Shuffle(Vector128<float> left, Vector128<float> right, byte control) { throw null; }
         public static Vector128<float> Sqrt(Vector128<float> value) { throw null; }
         public static Vector128<float> SqrtScalar(Vector128<float> value) { throw null; }
         public static unsafe void StoreAligned(float* address, Vector128<float> source) { throw null; }
         public static unsafe void StoreAlignedNonTemporal(float* address, Vector128<float> source) { throw null; }
         public static unsafe void Store(float* address, Vector128<float> source) { throw null; }
+        public static unsafe void StoreHigh(float* address, Vector128<float> source) { throw null; }
+        public static unsafe void StoreLow(float* address, Vector128<float> source) { throw null; }
```
Comment (tannergooding) [inline src/System.Runtime.Intrinsics.X86/ref/System.Runtime.Intrinsics.X86.cs]: StoreLow from Sse2 looks to be missing

Comment (fiigii) [reply src/System.Runtime.Intrinsics.X86/ref/System.Runtime.Intrinsics.X86.cs]: https://github.com/dotnet/corefx/blob/master/src/System.Runtime.Intrinsics.X86/ref/System.Runtime.Intrinsics.X86.cs#L1065

That was not missing, so it is not shown in this PR.

Comment (tannergooding) [reply src/System.Runtime.Intrinsics.X86/ref/System.Runtime.Intrinsics.X86.cs]: :+1:

**Comment by tannergooding** [comment]
PR should pass after https://github.com/dotnet/corefx/pull/26230 is merged

**Comment by benaadams** [comment]
@dotnet-bot test

**Comment by benaadams** [comment]
Missing coreclr item
```
MembersMustExist : Member 
'System.Runtime.Intrinsics.X86.Avx.ExtractByte(System.Runtime.Intrinsics.Vector256, System.Byte)' 
does not exist in the implementation but it does exist in the contract. 
```

**Comment by benaadams** [comment]
@dotnet-bot test CI

**Comment by benaadams** [comment]
@dotnet-bot test Windows x86 Release Build

**Comment by benaadams** [comment]
@dotnet-bot test Packaging All Configurations x64 Debug Build
@dotnet-bot test Alpine.3.6 x64 Debug Build
@dotnet-bot test Linux arm Release Build
@dotnet-bot test Linux x64 Release Build
@dotnet-bot test OSX x64 Debug Build
@dotnet-bot test Tizen armel Debug Build
@dotnet-bot test Windows x64 Debug Build

**Comment by fiigii** [comment]
Seems the CI does not detect the CoreCLR change yet.

**Comment by tannergooding** [comment]
@fiigii, the CoreCLR changes are in the PR I linked above (#26230). That PR is currently failing due to the APIs whose names have changed.

**Comment by benaadams** [comment]
So these two PRs are dependent on each other...

**Comment by tannergooding** [comment]
@benaadams, yup. Will likely require combining them, or some such.

**Comment by jkotas** [comment]
Cherry picked into https://github.com/dotnet/corefx/pull/26230

**Pull request closed without merging**